.github/workflows/phpstan.yml

```diff
@@ -0,0 +1,26 @@
+name: CakePHP CI
+
+on: [push, pull_request]
+
+jobs:
+  static-analysis:
+    name: Static Analysis
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Setup PHP
+        run: sudo phpdismod xdebug
+
+      - uses: actions/cache@v1
+        with:
+          path: ~/.composer/cache/files
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
```

Comment:
Usually this uses `composer.lock` but, as a library, we don't have such a file.

This also means _any_ change to `composer.json` will invalidate the installation cache. But given that it doesn't change very often, I'd say this is acceptable for a start.